src/parser/shaman/elemental/modules/talents/EchoingShock.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import { formatPercentage } from 'common/format';
+
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+
+import Events, { CastEvent } from 'parser/core/Events';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+
+class EchoingShock extends Analyzer {
+
+
+  badCasts = 0;
+  casts = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.ECHOING_SHOCK_TALENT.id);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER), this.onCast);
+    }
+
+
+  onCast(event: CastEvent) {
+    if(!this.selectedCombatant.hasBuff(SPELLS.ECHOING_SHOCK_TALENT.id)){
+      return;
+    }
+
+    this.casts += 1;
+
+    if(event.ability.guid !== SPELLS.LAVA_BURST.id){
```

Comment:
I'm a bit confused, this returns when the used spell was not Lava Burst, that means only Lava Bursts are counted towards `badCasts`.
The suggestions tells me to maximize Echoing Shock by using it **with** Lava Burst.

When I look at [this](https://www.warcraftlogs.com/reports/LVt29NWTBPjkYq7F#fight=53&type=casts&source=17&translate=true&view=events) log (just a random ranking from WCL) it seems like the player is following Echoing Shock with Earth Shock and Flame Shock, **not Lava Burst**. Is the suggestion-text wrong or am I missing something?

Reply:
Nah I am just dumb.